tests/libvmi/lifecycle.go

```diff
@@ -1,46 +1,121 @@
 package libvmi
 
 import (
+	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/client-go/kubecli"
 
 	"kubevirt.io/kubevirt/tests"
 )
 
-func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+type VMIPool map[string]*v1.VirtualMachineInstance
+
+func NewVMIPool() VMIPool {
+	return map[string]*v1.VirtualMachineInstance{}
+}
+
+func (vmis VMIPool) Setup(loggedInExpecterFactory tests.VMIExpecterFactory) {
 	virtClient, err := kubecli.GetKubevirtClient()
 	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
 
-	By("Creating the VMI")
-	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	vmis.create(loggedInExpecterFactory, virtClient)
+	vmis.waitUntilReady(loggedInExpecterFactory, virtClient)
```

Comment:
aint it a good practice that `create` would return the new created vmi , and here you will reassign it to the var, and use it on the next line ? this way you will have the updated scheme with the UID and so on, on PR #4000 it did problems,
i think its a good practice to get a fresh instance ? (maybe here its not needed, but still possible good practice)

Reply:
no need